server/web/api/register.py

```diff
@@ -0,0 +1,123 @@
+from functools import wraps
+import math
+from random import random, choice
+
+from flask import abort, make_response, request, Blueprint
+from flask_api import status
+from flask_jwt_extended import create_access_token, get_jwt_identity, jwt_required
+from flask_login import login_user
+from flask_wtf.csrf import CSRFError, generate_csrf
+from tans.tans import uniques
+from models import WebUser
+from app import csrf
+
+# FIXME sollten wir zusammen mit dem Web package iwann mal umbenennen
+api = Blueprint("api-web", __name__)
+
+
+@api.errorhandler(CSRFError)
+def handle_csrf_error(error: CSRFError):
+    status = error.response or 400
+    return make_response(({"error": error.description}, status))
+
+
+def admin_only(func):
+    @wraps(func)  # https://stackoverflow.com/a/64534085/11370741
+    @jwt_required()
+    def wrapper(*args, **kwargs):
+        identity = get_jwt_identity()
+        if identity != "admin":
+            abort(status.HTTP_401_UNAUTHORIZED)
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+@api.get("/csrf")
+def get_csrf():
+    return {"csrf_token": generate_csrf()}
+
+
+@api.post("/login")
+@csrf.exempt  # TODO: Remove
+def login():
+    # Extract data from request
+    try:
+        username = request.form["username"]
+        password = request.form["password"]
+    except KeyError:
+        return {
+            "error": "Missing username or password in request"
+        }, status.HTTP_400_BAD_REQUEST
+
+    # Get user object from database
+    user = WebUser.get_by_username(username)
+    if user is None:
+        return {"error": f"""User with user name '{username}' does not exist"""}, status.HTTP_401_UNAUTHORIZED
+
+    # Check password
+    if not user.check_password(password):
+        return {"error": "Incorrect password"}, status.HTTP_401_UNAUTHORIZED
+
+    login_user(user)
+    return {"token": create_access_token(identity="admin"), "username": username}, 200
+
+
+@api.get("/templates")
+@admin_only
+def get_templates():
+    return [
+        {"id": 10023, "name": "Busunfall", "players": 5},
+        {"id": 900323, "name": "Explosion im Wohnviertel", "players": 10},
+    ]
+
+
+@api.post("/scenario/start")
+@admin_only
+def start_scenario():
+    try:
+        id = request.form["id"]
+        # TODO: Create actual execution
+        return {
+            "id": math.floor(random() * 1000),
+        }
+    except KeyError:
+        return {"error": "Missing id in request"}, 400
+
+
+test_tans = [str(x) for x in uniques(5)]
+
+test_execution = {
+    "id": -1,
+    "status": "",
+    "players": [
+        {
+            "tan": x,
+            "name": "Max Mustermann",
+            "status": choice(["", "In Vorbereitung"]),
+            "action": "Legt Zugang"
+        } for x in test_tans]
+}
```

Comment:
Test data in an endpoint file. You can use the test-data defined in executions/run.py. If you still want to use ur data, i would recommend an extra file.

Reply:
This change seems unneccessary as the next step is to implement the actual backend. I would leave it as is.

Reply:
If you dont need the test data in the due to your next step the test data isnt required at all. Therefore you can delete it and keep your file clean.

Reply:
The test data is neccessary to see if the website is working properly. This outweighs the need to keep the file clean imo.

Reply:
If you need the test data, therefore I would recommend to create an extra file, because in the current file they are missplaced imo.